src/utils/addresses.ts

```diff
@@ -73,3 +77,18 @@ export const sortAddressList = (addresses: Address[]): Address[] =>
     if (b.settings.isMain) return 1
     return (b.lastUsed ?? 0) - (a.lastUsed ?? 0)
   })
+
+// See https://github.com/alephium/desktop-wallet/issues/236
+export const migrateAddressMetadata = () => {
+  const Storage = getStorage()
+  const walletNames = Storage.list()
+
+  for (const name of walletNames) {
+    const data = localStorage.getItem(`${name}-addresses-metadata`)
+
+    if (data) {
+      localStorage.setItem(constructMetadataKey(name), data)
+      localStorage.removeItem(`${name}-${addressesMetadataLocalStorageKeyPrefix}`)
```

Comment:
Why use once `${name}-addresses-metadata` and then `${name}-${addressesMetadataLocalStorageKeyPrefix}` to identify the old item?

Reply:
cause I'm an idiot

Reply:
I guess I brain farted on this one too because it didn't register in my head at all :upside_down_face:

Reply:
fixed